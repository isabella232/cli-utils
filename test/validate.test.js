const { validateFlag, validate } = require('../lib/validate');

const flagSchema = require('./fixtures/flags.json');

describe('validate', () => {
  test('validateFlag no type', () => {
    expect(validateFlag).toBeDefined();

    const result = validateFlag(flagSchema.help, true);
    expect(result).toBe(true);
  });

  test('validateFlag one type', () => {
    expect(validateFlag).toBeDefined();

    const result = validateFlag(flagSchema.content, 'batman');
    expect(result).toBe(true);
  });

  test('validateFlag many types', () => {
    expect(validateFlag).toBeDefined();

    let result = validateFlag(flagSchema.port, 8080);
    expect(result).toBe(true);

    result = validateFlag(flagSchema.port, '8080');
    expect(result).toBe(true);
  });

  test('validate all', () => {
    expect(validate).toBeDefined();
    const flags = flagSchema;
    const prefix = 'test';
    let argv = {};

    let result = validate({ argv, flags, prefix });
    expect(result).toBe(true);

    argv = { version: true };
    result = validate({ argv, flags, prefix });
    expect(result).toBe(true);

    argv = { devWare: {}, port: 8080 };
    result = validate({ argv, flags, prefix });
    expect(result).toBe(true);

    argv = { things: ['joker'] };
    result = validate({ argv, flags, prefix });
    expect(result).toBe(true);

    // alias for 'things' should be ignored
    argv = { batman: ['joker'] };
    result = validate({ argv, flags, prefix });
    expect(result).toBe(true);

    argv = { devWae: {} };
    result = validate({ argv, flags, prefix, throw: false });
    expect(result).toBe(false);

    argv = { devWare: true };
    let fn = () => validate({ argv, flags, prefix });
    expect(fn).toThrowErrorMatchingSnapshot();

    argv = { devWar: {} };
    fn = () => validate({ argv, flags, prefix });
    expect(fn).toThrowErrorMatchingSnapshot();
  });
});
