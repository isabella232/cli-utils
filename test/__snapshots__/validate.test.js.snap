// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate validate all 1`] = `
"Flags were specified with invalid values:

  [34m--dev-ware[39m  must be set to a [1mobject[22m

Please check the command executed."
`;

exports[`validate validate all 2`] = `
"Flags were specified that were not recognized:

  [34m--dev-war[39m  Did you mean [1m--dev-ware[22m?

Please check the command executed."
`;
